File: frontend/src/constants/case.ts
// 案件状态/类型枚举（与后端 backend/internal/constants/case.go 保持一致）
export const CaseStatus = {
  FILED: 'filed',
  INVESTIGATING: 'investigating',
  HEARING: 'hearing',
  CLOSED: 'closed',
  ARCHIVED: 'archived',
} as const

export const CaseStatusText: Record<string, string> = {
  [CaseStatus.FILED]: '已立案',
  [CaseStatus.INVESTIGATING]: '调查取证',
  [CaseStatus.HEARING]: '庭审中',
  [CaseStatus.CLOSED]: '已结案',
  [CaseStatus.ARCHIVED]: '已归档',
}

export const CaseType = {
  CIVIL: 'civil',
  CRIMINAL: 'criminal',
  ADMINISTRATIVE: 'administrative',
  COMMERCIAL: 'commercial',
  LABOR: 'labor',
} as const

export const CaseTypeText: Record<string, string> = {
  [CaseType.CIVIL]: '民事',
  [CaseType.CRIMINAL]: '刑事',
  [CaseType.ADMINISTRATIVE]: '行政',
  [CaseType.COMMERCIAL]: '商事',
  [CaseType.LABOR]: '劳动',
}

export const CaseStatusOptions = Object.entries(CaseStatusText).map(([value, label]) => ({ label, value }))
export const CaseTypeOptions = Object.entries(CaseTypeText).map(([value, label]) => ({ label, value }))

File: backend/internal/constants/case.go
package constants

// CaseStatus 案件状态枚举。
const (
	CaseStatusFiled         = "filed"
	CaseStatusInvestigating = "investigating"
	CaseStatusHearing       = "hearing"
	CaseStatusClosed        = "closed"
	CaseStatusArchived      = "archived"
)

// CaseStatusValues 全部案件状态值。
var CaseStatusValues = []string{CaseStatusFiled, CaseStatusInvestigating, CaseStatusHearing, CaseStatusClosed, CaseStatusArchived}

// CaseType 案件类型枚举。
const (
	CaseTypeCivil      = "civil"
	CaseTypeCriminal   = "criminal"
	CaseTypeAdmin      = "administrative"
	CaseTypeCommercial = "commercial"
	CaseTypeLabor      = "labor"
)

// CaseTypeValues 全部案件类型值。
var CaseTypeValues = []string{CaseTypeCivil, CaseTypeCriminal, CaseTypeAdmin, CaseTypeCommercial, CaseTypeLabor}

// IsValidCaseStatus 校验案件状态。
func IsValidCaseStatus(s string) bool {
	for _, v := range CaseStatusValues {
		if v == s {
			return true
		}
	}
	return false
}

// IsValidCaseType 校验案件类型。
func IsValidCaseType(s string) bool {
	for _, v := range CaseTypeValues {
		if v == s {
			return true
		}
	}
	return false
}

// UserRole 用户角色枚举。
const (
	RoleAdmin     = "admin"
	RoleLawyer    = "lawyer"
	RoleAssistant = "assistant"
)

// UserRoleValues 全部角色值。
var UserRoleValues = []string{RoleAdmin, RoleLawyer, RoleAssistant}
